test/client_ui_test.py

```diff
@@ -84,6 +85,82 @@ def test_draw():
     assert f.last == line.ljust(79) + "\n"
 
 
+def test_json_format(capsys):
+    fake_time = FakeTime()
+
+    # Size is unknown at this point.
+    pb = client.ProgressBar(phase="setting up", format="json", now=fake_time)
+    line_data = {
+        "progress": 0,
+        "description": "setting up",
+        "done": 0,
+        "elapsed": 0.0
+    }
+    line = f'{json.dumps(line_data)},'
+    assert capsys.readouterr().err == line + "\n"
+
+    # Size was updated, but no bytes were transferred yet.
+    fake_time.now += 0.1
+    pb.size = 3 * GiB
+    line_data["elapsed"] = 0.1
+    line = f'{json.dumps(line_data)},'
+    assert capsys.readouterr().err == line + "\n"
+
+    # Phase was updated.
+    fake_time.now += 0.2
+    pb.phase = "downloading image"
+    line_data["description"] = "downloading image"
+    line_data["elapsed"] = 0.3
+    line = f'{json.dumps(line_data)},'
+    assert capsys.readouterr().err == line + "\n"
+
+    # Write some data...
+    fake_time.now += 0.8
+    pb.update(512 * MiB)
+    line_data["progress"] = 16
+    line_data["done"] = 512 * MiB
+    line_data["elapsed"] = 1.10
```

Comment:
It will be much nicer to use:

    expected = {"elapsed": ..., "size": ..., "done": ...}